spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/JdbcLockRegistry.java

```diff
@@ -95,22 +118,24 @@ private String pathFor(String input) {
 
 	@Override
 	public void expireUnusedOlderThan(long age) {
-		Iterator<Entry<String, JdbcLock>> iterator = this.locks.entrySet().iterator();
 		long now = System.currentTimeMillis();
-		while (iterator.hasNext()) {
-			Entry<String, JdbcLock> entry = iterator.next();
-			JdbcLock lock = entry.getValue();
-			if (now - lock.getLastUsed() > age && !lock.isAcquiredInThisProcess()) {
-				iterator.remove();
-			}
+		synchronized (this.locks) {
+			this.locks.entrySet()
+					.removeIf(entry -> {
+						JdbcLock lock = entry.getValue();
+						return now - lock.getLastUsed() > age && !lock.isAcquiredInThisProcess();
+					});
 		}
 	}
 
 	@Override
 	public void renewLock(Object lockKey) {
 		Assert.isInstanceOf(String.class, lockKey);
 		String path = pathFor((String) lockKey);
-		JdbcLock jdbcLock = this.locks.get(path);
+		JdbcLock jdbcLock;
+		synchronized (this.locks) {
+			jdbcLock = this.locks.get(path);
```

Comment:
I think that synchronization is also necessary here.

I think there are two reasons

at first
At the time of get()
If you `Map.put()` on another thread, the order of the nodes stored inside the `LinkedHashMap` is changed and somtimes an error may occur(`ConcurrentModificationException`).

the second
Since the `accessOrder` of the current `LinkedHashMap` is set to `true`, if the size of the Map has reached the threshold
When `LinkedHashMap.get()` is called, the delete action will occur.

Reply:
OK. Convinced. Thanks.

So, fix `cacheCapacity` property, please, and I'll merge this.